src/type_conversions/rpc/transaction.rs

```diff
@@ -3,13 +3,38 @@ use crate::type_conversions::{ToEthers, ToReth};
 use ethers::types::{
     OtherFields, Transaction as EthersTransaction, TransactionReceipt as EthersTransactionReceipt,
 };
-use reth_primitives::AccessList;
+use reth_primitives::{AccessList, Signature as PrimitiveSignature, TxType as PrimitiveTxType};
 use reth_revm::primitives::ruint::Uint;
 use reth_rpc_types::{Signature, Transaction, TransactionReceipt};
 
 /// Transaction (ethers) -> (reth)
 impl ToReth<Transaction> for EthersTransaction {
     fn into_reth(self) -> Transaction {
+        let primitive_tx_type = match self.transaction_type.map(|t| t.as_u64()) {
+            None => PrimitiveTxType::Legacy,
+            Some(0) => PrimitiveTxType::EIP2930,
+            Some(1) => PrimitiveTxType::EIP1559,
```

Comment:
EIP2930 is type 1, EIP1559 is type 2
```suggestion
            Some(1) => PrimitiveTxType::EIP2930,
            Some(2) => PrimitiveTxType::EIP1559,
```